dts/bindings/spi/silabs,gecko-spi-eusart.yaml

```diff
@@ -0,0 +1,17 @@
+description: GECKO EUSART SPI
+
+compatible: "silabs,gecko-spi-eusart"
+
+include: [spi-controller.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  interrupts:
+    required: true
+
+  peripheral-id:
+    type: int
+    description: peripheral ID
+    required: true
```

Comment:
this is added because of HAL limitations, so a big -1

Reply:
> this is added because of HAL limitations, so a big -1

Grepping for `peripheral-id` under `dts/` shows lots of existing uses. Are all of those non-HAL related?

Reply:
The current bindings which specify it:
```
dts/bindings/clock/atmel,sam-pmc.yaml:  peripheral-id was defined as p-id. The p-id number should be consulted on
dts/bindings/clock/atmel,sam-pmc.yaml:  - peripheral-id
dts/bindings/dma/intel,sedi-dma.yaml:  peripheral-id:
dts/bindings/gpio/intel,sedi-gpio.yaml:  peripheral-id:
dts/bindings/gpio/silabs,gecko-gpio-port.yaml:  peripheral-id:
dts/bindings/i2c/intel,sedi-i2c.yaml:  peripheral-id:
dts/bindings/ipm/intel,sedi-ipm.yaml:  peripheral-id:
dts/bindings/serial/cypress,psoc6-uart.yaml:  peripheral-id:
dts/bindings/serial/intel,sedi-uart.yaml:  peripheral-id:
dts/bindings/serial/silabs,gecko-leuart.yaml:  peripheral-id:
dts/bindings/serial/silabs,gecko-uart.yaml:  peripheral-id:
dts/bindings/serial/silabs,gecko-usart.yaml:  peripheral-id:
dts/bindings/spi/cypress,psoc6-spi.yaml:  peripheral-id:
dts/bindings/spi/intel,sedi-spi.yaml:  peripheral-id:
dts/bindings/spi/silabs,gecko-spi-usart.yaml:  peripheral-id:
dts/bindings/spi/telink,b91-spi.yaml:  peripheral-id:
dts/bindings/watchdog/silabs,gecko-wdog.yaml:  peripheral-id:
```

Reply:
these are typically halisms

Reply:
> > this is added because of HAL limitations, so a big -1
> 
> Grepping for `peripheral-id` under `dts/` shows lots of existing uses. Are all of those non-HAL related?

@gmarull also, it would be helpful if you could provide a reference to the place in our documentation which defines this policy?

Reply:
You can find numerous PRs where this has been rejected. DT describes hardware, not properties that are purely there to do concatenations required by HAL usage.

cc: @henrikbrixandersen

Reply:
We have been rejecting DTS properties solely for HAL peripheral identifying purposes multiple times. 

Way back when, we used to have these in Kconfig, which was a mess. When we moved from device instances in Kconfig to devicetree, it was decided not to allow these kind of properties to avoid carrying over the mess from Kconfig. Not sure it is recorded anywhere, though. 

Is the peripheral ID used by the hardware in this case? E.g. is it written to a register or similar? Or is it solely used for mapping from a shim driver to the HAL functions?

Reply:
It's only used to be able to map to HAL functions in this case. I don't particularly care whether the property exists or if the driver does its own mapping based on peripheral base address, I just don't see a need for the driver to support _both_ (which the other Gecko UART and SPI drivers previously added by the community do).

Reply:
Agreed. I'd recommend going with the mapping in the driver, then. Either based on the base address or based on the devicetree nodelabel (we have multiple examples of this in-tree as well).

Reply:
In this particular case, it looks like you need proper `clocks` description in DT, where you can use a cell to provide a clock identifier. Other examples of this approach are what ambiq pwr does.